File: lib/boa/templates/objc/AppDependencies.m
//
//  <%= @project %>AppDependencies.m
//  <%= @project %>
//
//  Created by <%= @author %> on <%= @date %>.
//
//

#import "<%= @project %>AppDependencies.h"

#import "RootRouter.h"

@interface <%= @project %>AppDependencies ()

@end

@implementation <%= @project %>AppDependencies

- (id)initWithWindow:(UIWindow *)window
{
    if ((self = [super init]))
    {
        [self configureDependencies:window];
    }

    return self;
}

- (void)installRootViewController
{
    // *** present first Router here
}

- (void)configureDependencies:(UIWindow *)window
{
    // -----
    // root classes
    <%= @project %>RootRouter *rootRouter = [[<%= @project %>RootRouter alloc] initWithWindow:window];
    // *** add datastore

    // *** module initialization
}


@end

File: lib/boa/templates/swift/AppDependencies.swift
//
//  <%= @project %>AppDependencies.swift
//  <%= @project %>
//
//  Created by <%= @author %> on <%= @date %>.
//
//

import Foundation
import UIKit

class <%= @project %>AppDependencies: NSObject
{

    class func initWithWindow(window: UIWindow) -> <%= @project %>AppDependencies
    {

        let obj = <%= @project %>AppDependencies()
        obj.configureDependencies(window)

        return obj
    }

    func installRootViewController()
    {
        // *** present first Router here
    }

    func configureDependencies(window: UIWindow)
    {
        // -----
        // root classes
        let rootRouter = RootRouter.init(window: window)
        // *** add datastore

        // *** module initialization
    }
}
